demos/common/python/openvino/model_zoo/model_api/models/nanodet.py

```diff
@@ -0,0 +1,131 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import math
+import numpy as np
+
+from .types import NumericalValue, BooleanValue
+from .detection_model import DetectionModel
+from .utils import Detection, softmax, nms, clip_detections
+
+
+class NanoDet(DetectionModel):
+    __model__ = 'NanoDet'
+
+    def __init__(self, model_adapter, configuration=None, preload=False):
+        super().__init__(model_adapter, configuration, preload)
+        self._check_io_number(1, 1)
+        self.output_blob_name = self._get_outputs()
+        self.strides = [8, 16, 32] if self.is_legacy else [8, 16, 32, 64]
+        self.reg_max = 7
+
+    def _get_outputs(self):
+        output_blob_name = next(iter(self.outputs))
+        output_size = self.outputs[output_blob_name].shape
+        if len(output_size) != 3:
+            self.raise_error("Unexpected output blob shape {}. Only 3D output blob is supported".format(output_size))
+
+        return output_blob_name
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters['resize_type'].update_default_value('fit_to_window')
+        parameters['confidence_threshold'].update_default_value(0.05)
+        parameters.update({
+            'iou_threshold': NumericalValue(default_value=0.6, description="Threshold for NMS filtering"),
+            'num_classes': NumericalValue(default_value=80, description="Number of classes"),
```

Comment:
To prevent the error:
`[ ERROR ] Error with "num_classes" parameter:`
`[ ERROR ]       Incorrect value type <class 'int'>: should be <class 'float'>`
```suggestion
            'num_classes': NumericalValue(default_value=80, value_type=int, description="Number of classes"),
```

Reply:
Done